xds/src/main/java/io/grpc/xds/internal/sds/SdsProtocolNegotiators.java

```diff
@@ -138,6 +137,7 @@ public ChannelHandler newHandler(GrpcHttp2ConnectionHandler grpcHandler) {
       SslContextProviderSupplier localSslContextProviderSupplier =
           grpcHandler.getEagAttributes().get(XdsAttributes.ATTR_SSL_CONTEXT_PROVIDER_SUPPLIER);
       if (localSslContextProviderSupplier == null) {
+        checkNotNull(fallbackProtocolNegotiator, "fallbackProtocolNegotiator not set!");
```

Comment:
Should also mention the absence of `ATTR_SSL_CONTEXT_PROVIDER_SUPPLIER` attribute?

Reply:
That's implementation detail and should not be exposed to the user (who receives this exception). In this case the CDS policy didn't have any TLS config for the channel (cluster) and the user didn't provide a fallback-negotiator. Based on some other doc the user should know that fallback negotiator kicks in when xDS didn't provide TLS config.

Reply:
I agree `ATTR_SSL_CONTEXT_PROVIDER_SUPPLIER` is implementation details and should not be exposed to users. But   I feel it will be more informative if we mention something like "TLS config not available" in the first place before telling the user fallback is not provided, than just saying fallback is not set and expecting user to know from the doc why this is happening.

Reply:
Good point. Fixed the message in the check.